### ISSUE TITLE
Suggestion: Change the error type

### ISSUE THREAD

**Issue by sporto** [gege251/elm-validator-pipeline#1]
Hi, I like your package very much, but something that would prevent me from using it is the handling of errors as `String`
I want my errors to have custom types so I can do type safe things later.

Consider changing errors to a type provided by the user e.g.

```
type alias Validated err a =
    Result (List err) a

type Field
  = FieldName
  | FieldEmail
  | FieldPassword

Ok ValidatedForm
    |> validate (String.notEmpty (FieldName, "name is required")) form.name
    |> validate (String.isEmail (FieldEmail, "email is invalid")) form.email
    |> validateMany
        [ String.hasLetter (FieldPassword, "password needs to have letters")
        , String.hasNumber (FieldPassword, "password needs to have numbers")
        ]
        form.password
```

Thanks

**Comment by szg251** [comment]
@sporto  This is a great idea, and I thought about it myself after implementing it with String type.
I will definitely work on it, at will let you know, if there's some progress!

**Comment by szg251** [comment]
@sporto Hi! I just released 2.0.0 with parameterized error types. Check it out!
https://package.elm-lang.org/packages/gege251/elm-validator-pipeline/2.0.0/

**Comment by sporto** [comment]
@gege251 Great thanks

Maybe `Validator.Named` is unnecessary now. 
Because I can pass an error type like (Field, String). I can later find the errors I want by filtering by `Field`.
